src/cli-options.ts

```diff
@@ -0,0 +1,362 @@
+          message:
+            `Expected the WIREIT_PARALLEL env variable to be ` +
+            `a positive integer, got ${JSON.stringify(workerString)}`,
+          script,
+          type: 'failure',
+        },
+      };
+    }
+    return {ok: true, value: parsedInt};
+  })();
+  if (!numWorkersResult.ok) {
+    return numWorkersResult;
+  }
+
+  const cacheResult = ((): Result<'none' | 'local' | 'github'> => {
+    const str = process.env['WIREIT_CACHE'];
+    if (str === undefined) {
+      // The CI variable is a convention that is automatically set by GitHub
+      // Actions [0], Travis [1], and other CI (continuous integration)
+      // providers.
+      //
+      // [0] https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables
+      // [1] https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
+      //
+      // If we're on CI, we don't want "local" caching, because anything we
+      // store locally will be lost when the VM shuts down.
+      //
+      // We also don't want "github", because (even if we also detected that
+      // we're specifically on GitHub) we should be cautious about using up
+      // storage quota, and instead require opt-in via WIREIT_CACHE=github.
+      const ci = process.env['CI'] === 'true';
+      return {ok: true, value: ci ? 'none' : 'local'};
+    }
+    if (str === 'local' || str === 'github' || str === 'none') {
+      return {ok: true, value: str};
+    }
+    return {
+      ok: false,
+      error: {
+        reason: 'invalid-usage',
+        message:
+          `Expected the WIREIT_CACHE env variable to be ` +
+          `"local", "github", or "none", got ${JSON.stringify(str)}`,
+        script,
+        type: 'failure',
+      },
+    };
+  })();
+  if (!cacheResult.ok) {
+    return cacheResult;
+  }
+
+  const failureModeResult = ((): Result<FailureMode> => {
+    const str = process.env['WIREIT_FAILURES'];
+    if (!str) {
+      return {ok: true, value: 'no-new'};
+    }
+    if (str === 'no-new' || str === 'continue' || str === 'kill') {
+      return {ok: true, value: str};
+    }
+    return {
+      ok: false,
+      error: {
+        reason: 'invalid-usage',
+        message:
+          `Expected the WIREIT_FAILURES env variable to be ` +
+          `"no-new", "continue", or "kill", got ${JSON.stringify(str)}`,
+        script,
+        type: 'failure',
+      },
+    };
+  })();
+  if (!failureModeResult.ok) {
+    return failureModeResult;
+  }
+
+  if (process.argv[2] === 'watch') {
+    // TODO(aomarks) Remove in upcoming release.
+    return {
+      ok: false,
+      error: {
+        script,
+        type: 'failure',
+        reason: 'invalid-usage',
+        message:
+          `As of wireit v0.6, use "--watch" instead of "watch". ` +
+          `In an upcoming release, the "watch" argument will be passed to the script, ` +
+          `consistent with how npm usually behaves.`,
```

Comment:
Probably not a big deal for something meant to be temporary, but `watch` getting passed to script is only true for npm and gets swallowed as `unrecognized` for yarn and pnpm right?

Reply:
Yep, for yarn and pnpm it will actually be a warning about unrecognized arguments, unless they put it on the right-hand-side of the `--`. Only npm has the weird behavior where some arguments get forwarded and some turn into environment variables.

But I'm ok with writing the docs around npm behavior to keep things simple, since that's our primary target. It should be fairly clear to the yarn/pnpm user when this ultimately happens, since they will see the warning.